listing.php

```diff
@@ -445,14 +445,16 @@ function showTreeDir($svnrep, $path, $rev, $peg, $listing)
 	// For file, the last element in the subs is the file name.
 	// Therefore, it is always count($subs) - 2
 	$limit = count($subs) - 2;
+	$level = $limit;
+	$level = $level <= 0 ? 0 : $level;
```

Comment:
Why? Why not supply limit directly?

Reply:
Because conceptually the level to start access checks for and the limit to do whatever until are different things. Additionally, when I came across that problem, I needed a different level of `$limit - 1` for some reason. That doesn't seem to be the case for @k10blogger currently, but we don't know the reason yet and I don't have the time to check again in my setup currently. So if there's still some problem regarding the level, `blame` and the code as-is will make debugging easier.

Reply:
Alright, please add a comment why it is necessary not to collapse both statements.